pkg/controller/bootstrap/util_test.go

```diff
@@ -79,38 +81,46 @@ func TestValidateSecretForSigning(t *testing.T) {
 	}
 
 	for _, tc := range cases {
-		secret := &v1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace:       metav1.NamespaceSystem,
-				Name:            bootstrapapi.BootstrapTokenSecretPrefix + givenTokenID,
-				ResourceVersion: "1",
-			},
-			Type: bootstrapapi.SecretTypeBootstrapToken,
-			Data: map[string][]byte{
-				bootstrapapi.BootstrapTokenIDKey:           []byte(tc.tokenID),
-				bootstrapapi.BootstrapTokenSecretKey:       []byte(tc.tokenSecret),
-				bootstrapapi.BootstrapTokenUsageSigningKey: []byte(tc.okToSign),
-				bootstrapapi.BootstrapTokenExpirationKey:   []byte(tc.expiration),
-			},
-		}
-
-		tokenID, tokenSecret, ok := validateSecretForSigning(secret)
-		if ok != tc.valid {
-			t.Errorf("%s: Unexpected validation failure. Expected %v, got %v", tc.description, tc.valid, ok)
-		}
-		if ok {
-			if tokenID != tc.tokenID {
-				t.Errorf("%s: Unexpected Token ID. Expected %q, got %q", tc.description, givenTokenID, tokenID)
+		t.Run(tc.description, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
```

Comment:
`WithCancel` doesn't seem to be needed here.

Reply:
Done.